lumicks/pylake/nb_widgets/correlated_plot.py

```diff
@@ -0,0 +1,81 @@
+import numpy as np
+import warnings
+
+
+def plot_correlated(
+    channel_slice, frame_timestamps, get_plot_data, frame=0, reduce=np.mean, colormap="gray"
+):
+    """Downsample channel on a frame by frame basis and plot the results.
+
+    Parameters
+    ----------
+    channel_slice : pylake.channel.Slice
+        Data slice that we with to downsample.
+    frame_timestamps : list of tuple
+        List of tuples with start and stop timestamps of each frame.
+    get_plot_data : callable
+        Function that will return the plotdata for a frame.
+    frame : int
+        Frame to show.
+    reduce : callable
+        The function which is going to reduce multiple samples into one. The default is
+        :func:`numpy.mean`, but :func:`numpy.sum` could also be appropriate for some cases
+        e.g. photon counts.
+    colormap : str or Colormap
+        Colormap used for plotting.
+    """
+    import matplotlib.pyplot as plt
+
+    downsampled = channel_slice.downsampled_over(frame_timestamps, where="left", reduce=reduce)
+
+    if len(downsampled.timestamps) < len(frame_timestamps):
+        warnings.warn("Only subset of time range available for selected channel")
+
+    plot_data = get_plot_data(frame)
+    aspect_ratio = plot_data.shape[0] / np.max([plot_data.shape])
+
+    fig, (ax1, ax2) = plt.subplots(1, 2, figsize=plt.figaspect(aspect_ratio / 2))
+    t0 = downsampled.timestamps[0]
+    t, y = (downsampled.timestamps - t0) / 1e9, downsampled.data
```

Comment:
can we use `downsampled.seconds` to define `t` here?

Reply:
We certainly can. Fixing.